rmw_fastrtps_cpp/src/rmw_node.cpp

```diff
@@ -168,14 +168,23 @@ get_security_file_paths(
 
   const char * file_prefix = "file://";
 
-  std::string tmpstr;
+  char * file_path = nullptr;
   for (size_t i = 0; i < num_files; i++) {
-    tmpstr = std::string(rcutils_join_path(node_secure_root, file_names[i]));
-    if (!rcutils_is_readable(tmpstr.c_str())) {
-      return false;
+    file_path = rcutils_join_path(node_secure_root, file_names[i]);
+    if (file_path) {
+      if (!rcutils_is_readable(file_path)) {
+        free(file_path);
+        return false;
+      }
+
+      security_files_paths[i] = std::string(file_prefix + file_path);
```

Comment:
Oh, also, I'm pretty sure `file_prefix + file_path` is wrong here.  That's going to add two pointers, not concatenate the two of them together.

Reply:
exactly, the `std::string` should keep for `file_path` here